azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -0,0 +1,378 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/synapse/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSynapseSqlPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSynapseSqlPoolCreate,
+		Read:   resourceArmSynapseSqlPoolRead,
+		Update: resourceArmSynapseSqlPoolUpdate,
+		Delete: resourceArmSynapseSqlPoolDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				if _, err := parse.SynapseSqlPoolID(d.Id()); err != nil {
+					return []*schema.ResourceData{d}, err
+				}
+
+				d.Set("create_mode", "Default")
+				if v, ok := d.GetOk("create_mode"); ok && v.(string) != "" {
+					d.Set("create_mode", v)
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SynapseSqlPoolName,
+			},
+
+			"synapse_workspace_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SynapseWorkspaceID,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "DW1000c",
+				ValidateFunc: validation.StringInSlice([]string{
+					"DW100c",
+					"DW200c",
+					"DW300c",
+					"DW400c",
+					"DW500c",
+					"DW1000c",
+					"DW1500c",
+					"DW2000c",
+					"DW2500c",
+					"DW3000c",
+					"DW5000c",
+					"DW6000c",
+					"DW7500c",
+					"DW10000c",
+					"DW15000c",
+					"DW30000c",
+				}, false),
+			},
+
+			"create_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Default",
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Default",
+					"Recovery",
+					"PointInTimeRestore",
+				}, false),
+			},
+
+			"collation": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"source_database_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.Any(
+					validate.SynapseSqlPoolID,
+					sqlValidate.SqlDatabaseID,
+				),
+			},
+
+			"restore_point_in_time": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.IsRFC3339Time,
+			},
+
+			"data_encrypted": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
```

Comment:
this looks like a value that'll change in the future:

a) is this defaulted off?
b) is this going to change to `on` in the future?
c) is this not a ForceNew field like in many other API's?

Reply:
a) for now, it's default "off"
b) I am sorry that I don't know whether it will change default value in the future
c) it's a updatable field. I don't see similiar fields in other API?